src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/Context/FormattingContext.IndentationData.cs

```diff
@@ -81,11 +100,25 @@ public SyntaxToken EndToken
                 get { return this.Operation.EndToken; }
             }
 
-            public override int Indentation => _lazyIndentationDelta.Value + _baseIndentationGetter(_formattingContext, Operation);
+            private int GetOrComputeIndentationDelta()
+            {
+                var indentationDelta = Volatile.Read(ref _lazyIndentationDelta);
+                if (indentationDelta != UninitializedIndentationDelta)
+                    return indentationDelta;
+
+                indentationDelta = _indentationDeltaGetter(_formattingContext, Operation);
+                var existingIndentationDelta = Interlocked.CompareExchange(ref _lazyIndentationDelta, indentationDelta, UninitializedIndentationDelta);
+                if (existingIndentationDelta != UninitializedIndentationDelta)
+                    return existingIndentationDelta;
+
+                return indentationDelta;
```

Comment:
confused by this code.  very very confused by it.  needs comments majorly.  if this is just an int, it's unclear to me why we need volatile at all.

if this is to somehow allow multiple callers to initialize, but let first win, why not let any win?  

if this is to allow multiple callers to initialize, and they'll get different results... then 😨

Reply:
This is the standard optimistic lazy initialization pattern, but I've extracted a helper method so it's reusable.

Reply:
> ... where all callers will produce the same value ...

The code does not assume or require this.